src/GraphBLAS-sharp/Implementations.fs

```diff
@@ -115,26 +116,549 @@ type CSRMatrix<'a when 'a : struct and 'a : equality>(csrTuples: CSRFormat<'a>)
+
+    let mutable rows, columns, values = rows, columns, values
+    member this.Rows with get() = rows
+    member this.Columns with get() = columns
+    member this.Values with get() = values
+    member this.Elements with get() = (rows, columns, values) |||> Array.zip3
+
+    override this.Clear () = failwith "Not Implemented"
+    override this.Copy () = failwith "Not Implemented"
+    override this.Resize a b = failwith "Not Implemented"
+    override this.GetNNZ () = failwith "Not Implemented"
+
+    override this.GetTuples () =
+        opencl {
+            return {| Rows = this.Rows; Columns = this.Columns; Values = this.Values |}
+        }
+
+    override this.GetMask(?isComplemented: bool) =
+        let isComplemented = defaultArg isComplemented false
+        failwith "Not Implemented"
+
+    override this.ToHost () =
+        opencl {
+            let! _ = ToHost this.Rows
+            let! _ = ToHost this.Columns
+            let! _ = ToHost this.Values
+
+            return upcast this
+        }
+
+    override this.Extract (mask: Mask2D option) : OpenCLEvaluation<Matrix<'a>> = failwith "Not Implemented"
+    override this.Extract (colMask: Mask1D option * int) : OpenCLEvaluation<Vector<'a>> = failwith "Not Implemented"
+    override this.Extract (rowMask: int * Mask1D option) : OpenCLEvaluation<Vector<'a>> = failwith "Not Implemented"
+    override this.Extract (idx: int * int) : OpenCLEvaluation<Scalar<'a>> = failwith "Not Implemented"
+    override this.Assign (mask: Mask2D option, value: Matrix<'a>) : OpenCLEvaluation<unit> = failwith "Not Implemented"
+    override this.Assign (colMask: Mask1D option * int, value: Vector<'a>) : OpenCLEvaluation<unit> = failwith "Not Implemented"
+    override this.Assign (rowMask: int * Mask1D option, value: Vector<'a>) : OpenCLEvaluation<unit> = failwith "Not Implemented"
+    override this.Assign (idx: int * int, value: Scalar<'a>) : OpenCLEvaluation<unit> = failwith "Not Implemented"
+    override this.Assign (mask: Mask2D option, value: Scalar<'a>) : OpenCLEvaluation<unit> = failwith "Not Implemented"
+    override this.Assign (colMask: Mask1D option * int, value: Scalar<'a>) : OpenCLEvaluation<unit> = failwith "Not Implemented"
+    override this.Assign (rowMask: int * Mask1D option, value: Scalar<'a>) : OpenCLEvaluation<unit> = failwith "Not Implemented"
+
+    override this.Mxm a b c = failwith "Not Implemented"
+    override this.Mxv a b c = failwith "Not Implemented"
+
+    member internal this.EWiseAddCOO
+        (matrix: COOMatrix<'a>)
+        (mask: Mask2D option)
+        (semiring: Semiring<'a>) : OpenCLEvaluation<Matrix<'a>> =
+
+        let (BinaryOp append) = semiring.PlusMonoid.Append
+        let zero = semiring.PlusMonoid.Zero
+
+        //It is useful to consider that the first array is longer than the second one
+        let firstRows, firstColumns, firstValues, secondRows, secondColumns, secondValues, plus =
+            if this.Rows.Length > matrix.Rows.Length
```

Comment:
[Formatting conventions #1](https://docs.microsoft.com/ru-ru/dotnet/fsharp/style-guide/formatting#formatting-if-expressions)
[Formatting conventions #2](https://github.com/fsprojects/fantomas/blob/master/docs/FormattingConventions.md#conditional-expressions)

Reply:
@dpanfilyonok can we configure fslint to fail build in such cases?

Reply:
Yes.